tests/Unit/PrestaShopBundle/Translation/Provider/DefaultCatalogueProviderTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Unit\PrestaShopBundle\Translation\Provider;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShopBundle\Translation\Provider\DefaultCatalogueProvider;
+use PrestaShopBundle\Translation\Provider\TranslationCatalogueProviderInterface;
+use Symfony\Component\Translation\Dumper\XliffFileDumper;
+use Symfony\Component\Translation\MessageCatalogue;
+
+class DefaultCatalogueProviderTest extends TestCase
+{
+    /**
+     * @var string
+     */
+    private static $tempDir;
+
+    private static $wordings = [
+        'ShopSomeDomain' => [
+            'Some wording' => 'Some wording',
+            'Some other wording' => 'Some other wording',
+        ],
+        'ShopSomethingElse' => [
+            'Foo' => 'Foo',
+            'Bar' => 'Bar',
+        ],
+    ];
+
+    private static $emptyWordings = [
+        'ShopSomeDomain.en-US' => [
+            'Some wording' => '',
+            'Some other wording' => '',
+        ],
+        'ShopSomethingElse.en-US' => [
+            'Foo' => '',
+            'Bar' => '',
+        ],
+    ];
+
+    public static function setUpBeforeClass()
+    {
+        self::$tempDir = implode(DIRECTORY_SEPARATOR, [sys_get_temp_dir(), 'DefaultCatalogueProviderTest']);
```

Comment:
We already used `vfsStream` in other PHPUnit tests, maybe we can reuse it here?
See https://medium.com/swlh/phpunit-mocking-the-real-file-system-f6da6d1802cf